### PR TITLE
feat: migrate cache to posgre

### PR DESCRIPTION
**Pull request by KagChi**

### CODE REVIEW

**Review by Hazmi35 — CHANGES_REQUESTED** [review]
That, and:

I don't know what the exact difference between using relation() or use .references() directly

File: src/Listeners/Caches/Channels/ChannelCreateListener.ts [hunk 1]
```diff
@@ -17,10 +17,9 @@ export class ChannelCreateListener extends Listener {
 
     public async run(payload: { data: GatewayChannelCreateDispatch; shardId: number; }): Promise<void> {
         if (stateChannels) {
-            const key = "guild_id" in payload.data.d && payload.data.d.guild_id !== undefined ? GenKey(RedisKey.CHANNEL_KEY, payload.data.d.id, payload.data.d.guild_id) : GenKey(RedisKey.CHANNEL_KEY, payload.data.d.id);
-            const exists = await this.store.redis.exists(key);
-            await this.store.redis.set(key, JSON.stringify(payload.data.d));
-            if (exists === 0) await this.store.redis.incr(GenKey(RedisKey.CHANNEL_KEY, RedisKey.COUNT));
+            await this.store.drizzle.insert(channels).values({
+                id: payload.data.d.id
+            }).onConflictDoNothing({ target: channels.id });
```
[the remark this answers is not in the document]
Comment (KagChi) [reply src/Listeners/Caches/Channels/ChannelCreateListener.ts:22]: temporary

File: src/Structures/NezuGateway.ts [hunk 2]
```diff
@@ -55,8 +55,8 @@ export class NezuGateway extends EventEmitter {
             afk: false
         },
         updateSessionInfo: async (shardId: number, sessionInfo: SessionInfo | null) => {
-            if (gatewayResume && sessionInfo !== null) {
-                await this.drizzle.insert(schema.sessions).values({
+            await (gatewayResume && sessionInfo !== null
```
Comment (Hazmi35) [inline src/Structures/NezuGateway.ts:58]: Why the ternary?

Comment (KagChi) [reply src/Structures/NezuGateway.ts:58]: this is because of ur eslint :) it force me to use ternary

**Pull request merged**